feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/controller/set/SetControllerViewModel.kt

```diff
@@ -167,7 +167,8 @@ class SetControllerViewModel(
                 stashAddress = stashAddress(),
                 controllerAddress = controllerAddress,
                 fee = fee,
-                transferable = assetFlow.first().transferable
+                transferable = assetFlow.first().transferable,
+                controllerTransferable = stakingInteractor.assetFlow(controllerAddress).first().transferable
```

Comment:
Interactor only gives an asset flow for an address that is in our local database. As for the new controller, it may be an arbitrary address, so this approach won't work. My suggestion is to add a method to WalletRepository which will query remote storage for system.accountInfo and check its `free` balance (example can be found in `StakingRepositoryImpl.getControllerAccountInfo`). Then, you can use this method in validation directly without ViewModel be needed to pass it in the payload (it is more reasonable since payloads should contain only user input and some shared data, whereas we need this balance in one specific validation)